### PR TITLE
sql: add a lease test for #22479

### PR DESCRIPTION
**Pull request by knz**
This is meant as a test for #22440 but cannot be used do to a possibly unrelated bug #22479.

Proposing to merge this PR so that a subsequent investigator can troubleshoot the test.

Release note: none

### CODE REVIEW

File: pkg/sql/lease_test.go [hunk 1]
```diff
@@ -1048,3 +1048,125 @@ CREATE TABLE t.test (k CHAR PRIMARY KEY, v CHAR);
...
+			}
+		}
+		wg.Done()
+	}()
+
+	go func() {
+		leaseMgr := t.node(1)
+		for time.Now().Before(end) {
+			log.Infof(ctx, "publishing new descriptor")
+			desc, err := leaseMgr.Publish(ctx, descID, func(*sqlbase.TableDescriptor) error { return nil }, nil)
```
Comment (vivekmenezes) [inline pkg/sql/lease_test.go]: You have publishAndCheck() above. Why do you need this code?

Comment (knz) [reply pkg/sql/lease_test.go]: sorry the function above was a leftover from a previous attempt. cleaned up.

**Comment by vivekmenezes** [comment]
LGTM

**Comment by knz** [comment]
TFYR!

**Pull request merged**